From 3d877ca4561ab62fedc92e975015486031a6bf82 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Tue, 31 May 2016 11:58:23 +0300
Subject: [PATCH 1176/1240] dts: pcie: a80x0: Enable PCIe host on CP1

- Since the PCIe memory mapping for more than one CP overlaps
  each other, move the PCIe definitions from common CP110 DT
  to CP110 master and slave DTs and modify the memory windows
  according to Linux definitions for A8K devices.

Change-Id: I51588d7f0cad37e1d048b723256f39e95d494128
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30208
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/dts/cp110-master.dtsi | 25 +++++++++++++++++++++++++
 arch/arm/dts/cp110-slave.dtsi  | 25 +++++++++++++++++++++++++
 arch/arm/dts/cp110.dtsi        | 26 --------------------------
 3 files changed, 50 insertions(+), 26 deletions(-)

diff --git a/arch/arm/dts/cp110-master.dtsi b/arch/arm/dts/cp110-master.dtsi
index 6c860ce..0f17b28 100644
--- a/arch/arm/dts/cp110-master.dtsi
+++ b/arch/arm/dts/cp110-master.dtsi
@@ -32,6 +32,31 @@
 			#size-cells = <1>;
 
 			#include "cp110.dtsi"
+
+			pcie-controller {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "marvell,mvebu-dw-pcie";
+
+				pcie@0,0 {
+					reg = <0x00600000 0x20000>;
+					mem = <0xf6000000 0x1000000>;
+					cfg = <0xf6f00000 0x100000>;
+					status = "disabled";
+				};
+				pcie@1,0 {
+					reg = <0x00620000 0x20000>;
+					mem = <0xf7000000 0x1000000>;
+					cfg = <0xf7f00000 0x100000>;
+					status = "disabled";
+				};
+				pcie@2,0 {
+					reg = <0x00640000 0x20000>;
+					mem = <0xf8000000 0x1000000>;
+					cfg = <0xf8f00000 0x100000>;
+					status = "disabled";
+				};
+			};
 		};
 	};
 
diff --git a/arch/arm/dts/cp110-slave.dtsi b/arch/arm/dts/cp110-slave.dtsi
index 003ce27..ccc482e 100644
--- a/arch/arm/dts/cp110-slave.dtsi
+++ b/arch/arm/dts/cp110-slave.dtsi
@@ -31,6 +31,31 @@
 			#size-cells = <1>;
 
 			#include "cp110.dtsi"
+
+			pcie-controller {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "marvell,mvebu-dw-pcie";
+
+				pcie@0,0 {
+					reg = <0x00600000 0x20000>;
+					mem = <0xfa000000 0x1000000>;
+					cfg = <0xfaf00000 0x100000>;
+					status = "disabled";
+				};
+				pcie@1,0 {
+					reg = <0x00620000 0x20000>;
+					mem = <0xfb000000 0x1000000>;
+					cfg = <0xfbf00000 0x100000>;
+					status = "disabled";
+				};
+				pcie@2,0 {
+					reg = <0x00640000 0x20000>;
+					mem = <0xfc000000 0x1000000>;
+					cfg = <0xfcf00000 0x100000>;
+					status = "disabled";
+				};
+			};
 		};
 	};
 
diff --git a/arch/arm/dts/cp110.dtsi b/arch/arm/dts/cp110.dtsi
index a10472e..7d878bf 100644
--- a/arch/arm/dts/cp110.dtsi
+++ b/arch/arm/dts/cp110.dtsi
@@ -166,31 +166,6 @@ rtc-reg {
 	reg = <0x284000 104>;
 };
 
-pcie-controller {
-	#address-cells = <1>;
-	#size-cells = <1>;
-	compatible = "marvell,mvebu-dw-pcie";
-
-	pcie@0,0 {
-		reg = <0x00600000 0x20000>;
-		mem = <0xf6000000 0x1000000>;
-		cfg = <0xf6f00000 0x100000>;
-		status = "disabled";
-	};
-	pcie@1,0 {
-		reg = <0x00620000 0x20000>;
-		mem = <0xf7000000 0x1000000>;
-		cfg = <0xf7f00000 0x100000>;
-		status = "disabled";
-	};
-	pcie@2,0 {
-		reg = <0x00640000 0x20000>;
-		mem = <0xf8000000 0x1000000>;
-		cfg = <0xf8f00000 0x100000>;
-		status = "disabled";
-	};
-};
-
 comphy {
 	compatible = "marvell,mvebu-comphy", "marvell,comphy-cp110";
 	mux-bitcount = <4>;
@@ -230,4 +205,3 @@ ppv22@000000 {
 	};
 
 };
-
-- 
1.9.1

